### PR TITLE
fix(appsync): set a temporary variable for BinaryExpr inside an object put

### PR DESCRIPTION
**Pull request by brendo-m**
Fixes a weird but I ran into where `$obj.put('x', 1 + 1)` would cause a 500

### CODE REVIEW

File: src/vtl.ts [hunk 1]
```diff
@@ -99,11 +99,19 @@ export class VTL {
...
   public put(objVar: string, name: string, expr: string | Expr) {
-    this.qr(
-      `${objVar}.put(${name}, ${
-        typeof expr === "string" ? expr : this.eval(expr)
-      })`
-    );
+    let result;
+    if (typeof expr === "string") {
+      result = expr;
+    } else if (isBinaryExpr(expr)) {
```
Comment (sam-goodwin) [inline src/vtl.ts:105]: By the way, I think it's possible for this BinaryExpr to be an assignment where the operator is `=`

Are we handling this right?

Comment (brendo-m) [reply src/vtl.ts:105]: Wouldn't that be a `VariableStmt`?

**Comment by sam-goodwin** [comment]
Should this be fix(appsync) ?

File: src/vtl.ts (hunk 1 above)
Comment (sam-goodwin) [reply src/vtl.ts:105]: No, it would be an ExprtStmt(BinaryExpr) if it's assigning without const/let/var

```
a = b
```

Comment (sam-goodwin) [reply src/vtl.ts:105]: It's an expression that returns the value that is assigned. 

```ts
this.a = (a = b)
```

Comment (brendo-m) [reply src/vtl.ts:105]: Gross. You have to go out of your way to make it happen, but we are not handling it correctly.

```
// @ts-ignore
let y = 1;
return { x: (y = 2) };
```

Will fix in another PR

**Review by sam-goodwin — APPROVED** [review]

**Comment by thantos** [comment]
Is this true for other expressions too? 
```ts
{
   x : -1, // unary
   y: `template ${1}`,
   z: func() // some kind of call
}
```

**Comment by sam-goodwin** [comment]
Is this ready to be merged?

**Comment by brendo-m** [comment]
> Is this true for other expressions too?

Unary is broken right now because VTL doesn't like the space after the sign. `- 1` fails but `-1` is fine. Templates don't work either, not sure why. 

I'll merge this and create issues to follow up

**Pull request merged**